fees/eastern-neighborhoods-non-residential-open-space/viewmodel.js

```diff
@@ -0,0 +1,76 @@
+define([
+    'knockout',
+    'fees/abstract-fee',
+    'json!./settings.json',
+    './component'
+], function(ko, AbstractFee, settings) {
+    var EasternNeighborhoodsNonResidentialOpenSpaceFee = function(params) {
+        this.settings = settings;
+
+        AbstractFee.apply(this, [params]);
+
+        this.triggered = ko.computed(function() {
+            return this.isProjectInArea() &&
+                (
+                    this.newNonRes() > 0 ||
+                    this.newNonRes()/this.existingNonResGFA() >= .2 ||
+                    this.nonResGFA()/this.existingNonResGFA() >= .2 ||
+                    this.officeGFA()/this.existingNonResGFA() >= .2
+
+                );
+        }, this);
+
+        this.ready = ko.computed(function() {
+            if (!this.triggered()) {
+                return true;
+            }
+            return this.retailGFA() !== null && this.retailGFA() !== '' &&
+                this.retailOpenSpaceProvidedGFA() !== null && this.retailOpenSpaceProvidedGFA() !== '' &&
+                this.manufacturingGFA() !== null && this.manufacturingGFA() !== '' &&
+                this.manufacturingOpenSpaceProvidedGFA() !== null && this.manufacturingOpenSpaceProvidedGFA() !== '' &&
+                this.officeGFA() !== null && this.officeGFA() !== '' &&
+                this.officeOpenSpaceProvidedGFA() !== null && this.officeOpenSpaceProvidedGFA() !== '' &&
+                this.otherGFA() !== null && this.otherGFA() !== '' &&
+                this.otherOpenSpaceProvidedGFA() !== null && this.otherOpenSpaceProvidedGFA() !== '';
+        }, this);
+
+
+        this.calculatedFee = ko.computed(function() {
+            if (!this.triggered()) {
+                return 0;
+            }
+
+            this.calculateRequiredOpenSpaceOrFee = function(a, b, c) {
+                return (a-b)*eval(c);
+            };
+
+            this.retailOpenSpaceRequiredGFA(
```

Comment:
assigning values to observables inside of a computed is generally poor practice.  I would probably make these values their own computeds.